neuro_evolution_ctrnn/brain_visualizer/position.py

```diff
@@ -0,0 +1,123 @@
+        # testRay = numpy.array(testList)
+
+        ##### Create Graph from numpy Array
+        # G = nx.from_numpy_array(brainWeight)
+        #
+        # print(nx.info(G))
+        # print(G.edges(data=True))
+        # z = nx.get_edge_attributes(G, "weight")
+        # print("Attributes: ")
+
+        ##### Create Graph by adding Nodes and Edges seperate
+        G = nx.Graph(brain="CTRNN")
+
+        for i in range(len(brainState)):
+            G.add_node(i)
+
+        for zeile in range(len(brainState)):
+            for spalte in range(len(brainState)):
+                value = brainWeight[zeile, spalte]
+                G.add_edges_from([(zeile, spalte, {'myweight': value})])
+                # G.add_weighted_edges_from([(zeile, spalte, value)])
+
+        # print(nx.info(G))
+        # z = nx.get_edge_attributes(G, "myweight")
+        # print("Attributes: ")
+        # print(z)
+        # print(G.edges(data=True))
+
+        ######### Create initial Pose and write to Json file; If pose already in file, just read it
+        fpath = "position.json"
+        if os.path.isfile(fpath) and os.path.getsize(fpath) == 0:
+            numpyData = nx.spring_layout(G, k=5, weight="myweight", iterations=100)
+            with open("position.json", "w") as write_file:
+                json.dump(numpyData, write_file, cls=NumpyArrayEncoder)
+                write_file.close()
+
+        with open("position.json", "r") as read_file:
+            decodedArray = json.load(read_file)
+            read_file.close()
+        # print(decodedArray)
+        initialPosDict = {}
+        for i in range(30):
```

Comment:
die schleife verstehe ich nicht

Reply:
das ließt sich für mich so: decodedArray ist ein dict, in dem jedes object ein "i" attribute hat. Die objekte sollen dann nach ihrem "i" sortiert werden und in initialPosDict gespeichert werden.